storage/ceph.go

```diff
@@ -280,7 +277,31 @@ func (cluster *CephStorage) Put(poolname string, oid string, data io.Reader) (si
 				return 0, errors.New("Error wait_pending_front")
 			}
 		}
-		offset += uint64(len(bl))
+		offset += uint64(len(pending_data))
+
+		/* Resize current upload window */
+		expected_time := count * 1000 * 1000 * 1000 / current_upload_window  /* 1000 * 1000 * 1000 means use Nanoseconds */
+
+		// If the upload speed is less than half of the current upload window, reduce the upload window by half.
+		// If upload speed is larger than current window size per second, used the larger window and twice
+		if  (elapsed_time.Nanoseconds() / int64(expected_time)) > 2 {
```

Comment:
I think 
elapsed_time.Nanoseconds() > 2 * expected_time would be better.

Reply:
done